### ISSUE TITLE
Issue on Swiper on product image thumbnails

### ISSUE THREAD

**Issue by Kandax86** [Oksydan/modern-prestashop-starter-theme#106]
Hello,
i think i found an issue on your new management of thumbnails scrolling for product thumbs in product page,

you declare the galleryThumbs.on('afterInitSlider', ({ object, eventName }) => {

after initialisation of the object swiperslider so it never execute your event callback because in the instant of initialisation the thumb swiper there is no event on this emitter.

same issue for this one :
```
mainSlider.on('afterInitSlider', ({ object, eventName }) => {
        this.mainSliderSwiperInstance = mainSlider.SwiperInstance;
});
```

can you check?
thanks

**Comment by Kandax86** [comment]
i tried a poc on this bug

`class SliderEventEmitter {
  constructor() {
    this._events = {};
  }

  on(name, listener) {
    if (!this._events[name]) {
      this._events[name] = [];
    }

    this._events[name].push(listener);
  }

  emit(name, data) {
	  var emitter = this;
	  setTimeout(function(name, data){
    if (!emitter._events[name]) {
      return;
    }

    const fireCallbacks = (callback) => {
      callback(data);
    };

    emitter._events[name].forEach(fireCallbacks);
	  },10,name, data);
  }
}`

if i delayed for 10ms the emit of event it works thats why i think the event must be defined before the swiperslider creation.

**Comment by Oksydan** [comment]
Hi @Kandax86,

I didn't reproduce an issue but I am thinking about getting rid of events 😕.
We should replace it with awit after slideri jnitalization, I will be cleaner and easier to read. 

Instead of:

```javascript
const galleryThumbs = new prestashop.SwiperSlider(thumbsElem, {
      breakpoints: {
        320: {
          slidesPerView: 3,
        },
        576: {
          slidesPerView: 4,
        },
      },
      watchSlidesVisibility: true,
      watchSlidesProgress: true,
    })


    galleryThumbs.on('afterInitSlider', ({ object, eventName }) => {}):
    
```
There should be:

```javascript
    
    const galleryThumbs = new prestashop.SwiperSlider(thumbsElem, {
      breakpoints: {
        320: {
          slidesPerView: 3,
        },
        576: {
          slidesPerView: 4,
        },
      },
      watchSlidesVisibility: true,
      watchSlidesProgress: true,
    });
    
    /// galleryThumbsSwiper will be SwiperSlider instance.
    const galleryThumbsSwiper = await galleryThumbs.initSlider():
    
```
    
I think it will be introduced with version 2.3.0. I am not happy with that API is changing a lot lately but I think that change is required.

**Comment by Oksydan** [comment]
Hi @Kandax86,

I modified way of usage SwiperSlider class and how to get SwiperInstance properly [here](https://github.com/Oksydan/modern-prestashop-starter-theme/commit/012cfc5cec3888c1443bf87d4778dbb1976be9e6).

Documentation has been update accordingly to changes with SwiperSlider.

Changed is atm at develop branch so you can test it yourself. 
It will be available released with version 2.3.0.

**Comment by Kandax86** [comment]
thanks
